ckanext/datastore/plugin.py

```diff
@@ -265,6 +268,28 @@ def before_show(self, resource_dict):
 
         return resource_dict
 
+    def after_delete(self, context, resources):
+        model = context['model']
+        pkg = context['package']
+        res_query = model.Session.query(model.Resource)
+        query = res_query.filter(
+            model.Resource.package_id == pkg.id,
+            model.Resource.state == State.DELETED
+        )
+        deleted = [
+            res for res in query.all()
+            if res.extras.get('datastore_active') is True]
+
+        for res in deleted:
+            try:
+                logic.get_action('datastore_delete')(
+                    context, {'resource_id': res.id, 'force': True})
```

Comment:
We're inside the datastore code here, why not call the lower-level function to delete instead of the action?